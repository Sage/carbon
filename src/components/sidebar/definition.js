import { Sidebar } from './';
import DemoHelper from '../../utils/helpers/demo-helper';
import OptionsHelper from '../../utils/helpers/options-helper';
import podDefinition from '../pod/definition';

let definition = {
  component: Sidebar,
  key: 'sidebar',
  text: {
    bemClass: 'carbon-sidebar',
    details: '[content needed] Basic designs description for the component',
    description: '[content needed] Basic example of the component',
    name: 'Sidebar',
    type: 'modal'
  },
  defaultProps: Sidebar.defaultProps,
  props: Sidebar.propTypes,
  propOptions: {
    position: OptionsHelper.alignBinary(),
    size: OptionsHelper.sizesRestricted()
  }
};

definition.demoProps = {
  children: DemoHelper.elemArray(podDefinition, 7, 'title'),
  enableBackgroundUI: true,
  open: false,
  position: 'right',
  size: 'small'
};

export default definition;
