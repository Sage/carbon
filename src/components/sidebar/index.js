export { default } from "./sidebar.component";
