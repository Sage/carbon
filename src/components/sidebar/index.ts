export { default } from "./sidebar.component";
export type { SidebarProps } from "./sidebar.component";
