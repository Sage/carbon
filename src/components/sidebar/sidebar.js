import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import SidebarHeader from './sidebar-header';
import Icon from '../icon';
import Modal from '../modal';

/**
 * A Sidebar widget.
 *
 * == How to use a Sidebar in a component:
 *
 * In your file
 *
 *   import { Sidebar } from 'carbon-react/lib/components/sidebar';
 *
 * To render the Sidebar:
 *
 *   <Sidebar
 *     onCancel={ closeSidebar }
 *     open={ true }
 *   />
 *
 * Side bar is positioned on the right hand screen of the window by default.
 * To position the sidebar on the left hand side pass `position='left'` to the component.
 *
 * The background behind the sidebar is disabled by default. To allow the user to interact
 * with all the UI pass `enableBackgroundUI={ true}` to the component
 *
 * @class Sidebar
 * @constructor
 */
class Sidebar extends Modal {
  static propTypes = {

    /**
     * A custom close event handler
     *
     * @property onCancel
     * @type {Function}
     */
    onCancel: PropTypes.func,

    /**
     * Sets the open state of the sidebar
     *
     * @property open
     * @type {Boolean}
     * @default false
     */
    open: PropTypes.bool,

    /**
     * Determines if the user can interact with
     * the background ui when the sidebar is open
     *
     * @property enableBackgroundUI
     * @type {Boolean}
     * @default false
     */
    enableBackgroundUI: PropTypes.bool,

    /**
     * Determines the position of the sidebar
     * 'left' or 'right'
     *
     * @property position
     * @type {String}
     * @default 'right'
     */
    position: PropTypes.string,

    /**
     * Size of dialog, default size is 450px
     *
     * @property size
     * @type {String}
     * @default medium
     */
    size: PropTypes.string
  }

  static defaultProps = {
    position: 'right',
    size: 'medium'
  }

  /**
   * Returns classes for the component.
   *
   * @method mainClasses
   * @return {String} Main className
   */
  get mainClasses() {
    return classNames(
      'carbon-sidebar',
      this.props.className
    );
  }

  /**
   * Returns classes for the sidebar.
   *
   * @method sidebarClasses
   * @return {String} sidebar className
   */
  get sidebarClasses() {
    return classNames(
      'carbon-sidebar__sidebar',
      `carbon-sidebar__sidebar--${this.props.position}`,
      `carbon-sidebar__sidebar--${this.props.size}`
    );
  }

  /**
   * Returns the markup for the close icon.
   *
   * @method closeButton
   * @return {Object} JSX
   */
  get closeButton() {
    if (this.props.onCancel) {
      return (
        <span className='carbon-sidebar__close'>
          <Icon
            className='carbon-sidebar__close-icon'
            data-element='close'
            onClick={ this.props.onCancel }
            type='close'
          />
        </span>
      );
    }
    return null;
  }

  componentTags(props) {
    return {
      'data-component': 'sidebar',
      'data-element': props['data-element'],
      'data-role': props['data-role']
    };
  }

  /**
   * Returns the computed HTML for the sidebar.
   *
   * @method sidebarHTML
   * @return {Object} JSX for sidebar
   */
  get modalHTML() {
    return (
      <div
        className={ this.sidebarClasses }
        { ...this.componentTags(this.props) }
      >
        { this.closeButton }
        { this.props.children }
      </div>
    );
  }

  get transitionName() {
    return `sidebar--${this.props.position}`;
  }
}

export {
  Sidebar,
  SidebarHeader
};
