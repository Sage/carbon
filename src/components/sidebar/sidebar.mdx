import { Meta, ArgTypes, Canvas } from "@storybook/blocks";
import TranslationKeysTable from "../../../.storybook/utils/translation-keys-table";

import * as SidebarStories from "./sidebar.stories.tsx";

<Meta title="Sidebar" of={SidebarStories} />

# Sidebar

<a
  target="_blank"
  href="https://zeroheight.com/2ccf2b601/p/925533-sidebar"
  style={{ color: "#007E45", fontWeight: "bold", textDecoration: "underline" }}
  rel="noreferrer"
>
  Product Design System component
</a>

## Contents

- [Quick Start](#quick-start)
- [Related Components](#related-components)
- [Examples](#examples)
- [Props](#props)

## Quick Start

```javascript
import Sidebar from "carbon-react/lib/components/sidebar";
```

- Sidebar is positioned on the right hand screen of the window by default. To position the sidebar on the left hand side pass `position='left'` to the component.

- The background behind the sidebar is disabled by default. To allow the user to interact with all the UI pass `enableBackgroundUI={ true }` to the component

## Related Components

- Simple task in context? [Try Dialog](https://carbon.sage.com/?path=/docs/dialog--default-story "Try Dialog").
- Complex task that needs more space? [Try Dialog Full Screen](https://carbon.sage.com/?path=/docs/dialog-full-screen--default "Try Dialog Full Screen").

## Examples

### Default

<Canvas of={SidebarStories.DefaultStory} />

### Custom padding around content

<Canvas of={SidebarStories.CustomPaddingAroundContent} />

### With header

<Canvas of={SidebarStories.WithHeader} />

### With scroll

If sidebar content does not fit the sidebar, scroll will appear. Default scroll theme is "light".

<Canvas of={SidebarStories.WithScroll} />

### With Typography

<Canvas of={SidebarStories.WithTypography} />

### Allowing other elements to be focused

Using the `focusableContainers` prop allows elements that are not DOM children of the Sidebar to be reachable with the keyboard while the Sidebar is open.
This may occasionally be useful with things like Toasts where they persist on the page until dismissed by clicking a close icon.
(In the example below, try clicking the cancel and/or save buttons and notice how you can navigate across one or both toasts as well as the sidebar using the keyboard.)

<Canvas of={SidebarStories.OtherFocusableContainers} />

### Custom width

It is possible to set a custom width for the `Sidebar` via the `width` [prop](#props).
Setting this prop will override the preset width value defined via the `size` prop.

<Canvas of={SidebarStories.CustomWidth} />

### With header and footer padding overridden

<Canvas of={SidebarStories.WithHeaderAndFooterPadding} />

### Top modal override

When multiple modals are open on a page Carbon manages the order internally so that the last one mounted in the DOM
is treated as the top one and the focus trap is set on it. However, this can potentially
create unexpected issues if the loading of the dialogs isn't synchronous. In order to guarantee that a given
modal is treated as top set the `topModalOverride` prop. Please note that in the scenario where this prop is
set on multiple modals, the last one mounted with it will be treated as top.

<Canvas of={SidebarStories.TopModalOverride} />

## Props

### Sidebar

<ArgTypes of={SidebarStories} />

## Translation keys

The following keys are available to override the translations for this component by passing in a custom locale object to the
[i18nProvider](../?path=/docs/documentation-i18n--docs).

<TranslationKeysTable
  translationData={[
    {
      name: "sidebar.ariaLabels.close",
      description: "The text for close button aria-label attribute",
      type: "func",
      returnType: "string",
    },
  ]}
/>
