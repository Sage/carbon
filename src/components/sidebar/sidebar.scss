$sidebar-animation-length: 250ms;

// sidebar right animation:
.sidebar--right-enter {
  opacity: 0;
  right: -30px;
}

.sidebar--right-enter.sidebar--right-enter-active {
  opacity: 1;
  right: 0;
  transition: all $sidebar-animation-length ease-in;
}

.sidebar--right-exit {
  right: 0;
  opacity: 1;
}

.sidebar--right-exit.sidebar--right-exit-active {
  opacity: 0;
  right: -50px;
  transition: all $sidebar-animation-length ease-in;
}

// sidebar left animation:
.sidebar--left-enter {
  opacity: 0;
  left: -30px;
}

.sidebar--left-enter.sidebar--left-enter-active {
  opacity: 1;
  left: 0;
  transition: all $sidebar-animation-length ease-in;
}

.sidebar--left-exit {
  left: 0;
  opacity: 1;
}

.sidebar--left-exit.sidebar--left-exit-active {
  opacity: 0;
  left: -50px;
  transition: all $sidebar-animation-length ease-in;
}
