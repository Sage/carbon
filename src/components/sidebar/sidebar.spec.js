import React from 'react';
import 'jest-styled-components';
import { mount } from 'enzyme';
import Sidebar from './sidebar.component';
import Textbox from '../../__deprecated__/components/textbox';
import { SidebarStyle, SidebarCloseStyle } from './sidebar.style';
import { assertStyleMatch } from '../../__spec_helper__/test-utils';
import classicTheme from '../../style/themes/classic';

describe('Sidebar', () => {
  let wrapper, spy, preventDefault;

  beforeEach(() => {
    spy = jasmine.createSpy();
    preventDefault = jest.fn();
    wrapper = mount(
      <Sidebar
        open
        title='Test'
        className='custom-class'
        data-role='baz'
        data-element='bar'
        onCancel={ spy }
      >
        <Textbox />
        <Textbox />
        <Textbox />
      </Sidebar>
    );
  });

  describe('render', () => {
    describe('when sidebar is closed', () => {
      it('sets all the correct classes', () => {
        wrapper = mount(<Sidebar onCancel={ spy } />);
        expect(wrapper.find('div[data-component="sidebar"]').text()).toEqual('');
      });
    });

    describe('when onCancel is fired', () => {
      it('should remove focus', () => {
        jest.useFakeTimers();
        const tempDiv = document.createElement('div');
        const element = document.body.appendChild(tempDiv);

        wrapper = mount(
          <Sidebar open onCancel={ jest.fn }>
            <button type='button'>test content</button>
          </Sidebar>, { attachTo: element }
        );

        jest.runAllTimers();

        expect(document.activeElement).toMatchObject(wrapper.find('button'));
        wrapper.setProps({ open: false });
        expect(document.activeElement).toMatchObject(document.body);

        jest.clearAllTimers();
      });
    });

    describe('when enableBackgroundUI is enabled', () => {
      it('sets all the correct classes', () => {
        wrapper = mount(
          <Sidebar
            open
            enableBackgroundUI
            size='small'
            position='left'
            onCancel={ spy }
          />
        );
        expect(wrapper.find('[data-element="modal-background"]').length).toEqual(0);
      });
    });

    describe('when there is no onCancel prop', () => {
      it('should not have a close button', () => {
        wrapper = mount(
          <Sidebar
            open
          />
        );
        expect(wrapper.find('.carbon-sidebar__close').length).toEqual(0);
      });
    });
  });

  describe('Behaviour', () => {
    describe('clicking the close icon sidebar', () => {
      it('closes the sidebar', () => {
        const icon = wrapper.find('.carbon-sidebar__close-icon');
        icon.at(0).simulate('click');
        expect(spy).toHaveBeenCalled();
      });
    });

    describe('pressing Enter key on the close icon sidebar', () => {
      it('closes the sidebar', () => {
        const icon = wrapper.find('.carbon-sidebar__close-icon');
        icon.at(0).props().onKeyDown({ which: 13, preventDefault });
        expect(spy).toHaveBeenCalled();
      });
    });

    describe('pressing other key than Enter or Space', () => {
      it('does not close the sidebar', () => {
        const icon = wrapper.find('.carbon-sidebar__close-icon');
        icon.at(0).props().onKeyDown({ which: 16 });
        expect(spy).not.toHaveBeenCalled();
      });
    });
  });
});

describe('SidebarStyle', () => {
  describe('when prop size is passed to the component and position is set to right', () => {
    const wrapper = mount(<SidebarStyle
      open size='extra-small'
      position='right'
    />);

    it('should render correct style', () => {
      assertStyleMatch({
        width: '150px',
        boxShadow: '0 10px 30px 0 rgba(0,20,29,0.1),0 30px 60px 0 rgba(0,20,29,0.1)',
        right: '0'
      }, wrapper);
    });
  });

  describe('when prop left is passed to the component', () => {
    const wrapper = mount(<SidebarStyle
      open size='extra-small'
      position='left'
    />);

    it('should render correct style', () => {
      assertStyleMatch({
        boxShadow: '0 10px 30px 0 rgba(0,20,29,0.1),0 30px 60px 0 rgba(0,20,29,0.1)',
        left: '0'
      }, wrapper);
    });
  });

  describe('When classic style is passed to the component', () => {
    const closeIconWrapper = mount(<SidebarCloseStyle theme={ classicTheme } />);
    let wrapper;

    it('should render correct style', () => {
      wrapper = mount(<SidebarStyle
        theme={ classicTheme }
        open size='extra-small'
        position='left'
      />);

      assertStyleMatch({
        backgroundColor: '#e6ebed',
        borderRadius: '1px',
        bottom: '0',
        position: 'fixed',
        top: '0',
        padding: '20px',
        zIndex: '1002'
      }, wrapper);

      assertStyleMatch({
        color: 'rgba(0,0,0,0.85)',
        position: 'absolute',
        right: '20px',
        top: '15px',
        zIndex: '1'
      }, closeIconWrapper);
    });

    describe('when classic style is passed to the component and position is right', () => {
      wrapper = mount(<SidebarStyle theme={ classicTheme } position='right' />);

      assertStyleMatch({
        borderLeft: '1px solid #ccd6db'
      }, wrapper);
    });
  });
});
