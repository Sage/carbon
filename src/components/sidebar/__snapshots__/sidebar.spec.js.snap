// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar sidebarClasses returns a base sidebar 1`] = `
<Portal
  key="1"
>
  <div
    className="carbon-sidebar custom-class"
    data-component="sidebar"
    data-element="bar"
    data-role="baz"
    data-state="open"
  >
    <CSSTransitionGroup
      component="div"
      transitionAppear={true}
      transitionAppearTimeout={500}
      transitionEnter={true}
      transitionEnterTimeout={500}
      transitionLeave={true}
      transitionLeaveTimeout={500}
      transitionName="modal-background"
    >
      <div
        className="carbon-modal__background"
      />
    </CSSTransitionGroup>
    <CSSTransitionGroup
      component="div"
      transitionAppear={true}
      transitionAppearTimeout={500}
      transitionEnter={true}
      transitionEnterTimeout={500}
      transitionLeave={true}
      transitionLeaveTimeout={500}
      transitionName="sidebar--right"
    >
      <styled.div
        data-component="sidebar"
        data-element="bar"
        data-role="baz"
        position="right"
        size="medium"
        theme={
          Object {
            "colors": Object {
              "base": "#00A376",
              "border": "#668491",
              "brand": "#00DC00",
              "destructive": Object {
                "hover": "#9F2C3F",
              },
              "error": "#C7384F",
              "focus": "#FFB500",
              "focusedIcon": "#335B6D",
              "focusedLinkBackground": "#FFDA7F",
              "info": "#0073C2",
              "primary": "#26A826",
              "secondary": "#006300",
              "success": "#00B000",
              "tertiary": "#004400",
              "warning": "#E96400",
              "white": "#FFFFFF",
              "whiteMix": "#E5F4E5",
            },
            "disabled": Object {
              "background": "#E5EAEC",
              "border": "#CCD6DA",
              "button": "#E5EAEC",
              "buttonText": "rgba(0,0,0,.2)",
              "disabled": "rgba(0,0,0,0.55)",
              "input": "#F2F4F5",
              "text": "rgba(0,0,0,0.3)",
            },
            "icon": Object {
              "focus": "#335B6D",
            },
            "name": "base",
            "shadows": Object {
              "cards": "0 3px 3px 0 rgba(0,20,29,0.2), 0 2px 4px 0 rgba(0,20,29,0.15)",
              "depth1": "0 5px 5px 0 rgba(0,20,29,0.2), 0 10px 10px 0 rgba(0,20,29,0.1)",
              "depth2": "0 10px 20px 0 rgba(0,20,29,0.2), 0 20px 40px 0 rgba(0,20,29,0.1)",
              "depth3": "0 10px 30px 0 rgba(0,20,29,0.1), 0 30px 60px 0 rgba(0,20,29,0.1)",
              "depth4": "0 10px 40px 0 rgba(0,20,29,0.04), 0 50px 80px 0 rgba(0,20,29,0.1)",
            },
            "text": Object {
              "color": "rgba(0,0,0,0.9)",
              "placeholder": "rgba(0,0,0,0.3)",
              "size": "14px",
            },
          }
        }
      >
        <styled.div
          theme={
            Object {
              "colors": Object {
                "base": "#00A376",
                "border": "#668491",
                "brand": "#00DC00",
                "destructive": Object {
                  "hover": "#9F2C3F",
                },
                "error": "#C7384F",
                "focus": "#FFB500",
                "focusedIcon": "#335B6D",
                "focusedLinkBackground": "#FFDA7F",
                "info": "#0073C2",
                "primary": "#26A826",
                "secondary": "#006300",
                "success": "#00B000",
                "tertiary": "#004400",
                "warning": "#E96400",
                "white": "#FFFFFF",
                "whiteMix": "#E5F4E5",
              },
              "disabled": Object {
                "background": "#E5EAEC",
                "border": "#CCD6DA",
                "button": "#E5EAEC",
                "buttonText": "rgba(0,0,0,.2)",
                "disabled": "rgba(0,0,0,0.55)",
                "input": "#F2F4F5",
                "text": "rgba(0,0,0,0.3)",
              },
              "icon": Object {
                "focus": "#335B6D",
              },
              "name": "base",
              "shadows": Object {
                "cards": "0 3px 3px 0 rgba(0,20,29,0.2), 0 2px 4px 0 rgba(0,20,29,0.15)",
                "depth1": "0 5px 5px 0 rgba(0,20,29,0.2), 0 10px 10px 0 rgba(0,20,29,0.1)",
                "depth2": "0 10px 20px 0 rgba(0,20,29,0.2), 0 20px 40px 0 rgba(0,20,29,0.1)",
                "depth3": "0 10px 30px 0 rgba(0,20,29,0.1), 0 30px 60px 0 rgba(0,20,29,0.1)",
                "depth4": "0 10px 40px 0 rgba(0,20,29,0.04), 0 50px 80px 0 rgba(0,20,29,0.1)",
              },
              "text": Object {
                "color": "rgba(0,0,0,0.9)",
                "placeholder": "rgba(0,0,0,0.3)",
                "size": "14px",
              },
            }
          }
        >
          <Icon
            bgSize="small"
            className="carbon-sidebar__close-icon"
            data-element="close"
            onClick={[Function]}
            type="close"
          />
        </styled.div>
        <Textbox />
        <Textbox />
        <Textbox />
      </styled.div>
    </CSSTransitionGroup>
  </div>
</Portal>
`;

exports[`SidebarStyle When classic style is passed to the component should render correct style 1`] = `
.c0 {
  background-color: #1e499f;
  border-radius: 1px;
  bottom: 0;
  position: fixed;
  top: 0;
  padding: 27px 32px 32px 32px;
  z-index: 1002;
  width: 150px;
  box-shadow: 0 10px 30px 0 rgba(0,20,29,0.1),0 30px 60px 0 rgba(0,20,29,0.1);
  left: 0;
  background-color: #e6ebed;
  padding: 20px;
  border-right: 1px solid #ccd6db;
  box-shadow: 10px 0 15px rgba(0,0,0,.05);
  left: 0;
}

<styled.div
  open={true}
  position="left"
  size="extra-small"
  theme={
    Object {
      "colors": Object {
        "base": "#00A376",
        "baseBlue": "#255BC7",
        "border": "#668491",
        "brand": "#00DC00",
        "destructive": Object {
          "hover": "#9F2C3F",
        },
        "error": "#C7384F",
        "focus": "#FFB500",
        "focusedIcon": "#335B6D",
        "focusedLinkBackground": "#FFDA7F",
        "greyDarkBlue50": "#8099a4",
        "info": "#0073C2",
        "primary": "#00805D",
        "secondary": "#006045",
        "success": "#00B000",
        "tertiary": "#003F2E",
        "warning": "#FFB500",
        "white": "#FFFFFF",
        "whiteMix": "#E5F4E5",
      },
      "disabled": Object {
        "background": "#E5EAEC",
        "border": "#4d7080",
        "button": "#E5EAEC",
        "buttonText": "rgba(0,0,0,.2)",
        "disabled": "#b3c2c8",
        "input": "#1e499f",
        "text": "#003349",
      },
      "icon": Object {
        "focus": "#335B6D",
      },
      "name": "classic",
      "shadows": Object {
        "cards": "0 3px 3px 0 rgba(0,20,29,0.2), 0 2px 4px 0 rgba(0,20,29,0.15)",
        "depth1": "0 5px 5px 0 rgba(0,20,29,0.2), 0 10px 10px 0 rgba(0,20,29,0.1)",
        "depth2": "0 10px 20px 0 rgba(0,20,29,0.2), 0 20px 40px 0 rgba(0,20,29,0.1)",
        "depth3": "0 10px 30px 0 rgba(0,20,29,0.1), 0 30px 60px 0 rgba(0,20,29,0.1)",
        "depth4": "0 10px 40px 0 rgba(0,20,29,0.04), 0 50px 80px 0 rgba(0,20,29,0.1)",
      },
      "text": Object {
        "color": "rgba(0,0,0,0.9)",
        "placeholder": "rgba(0,0,0,0.3)",
        "size": "14px",
      },
    }
  }
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  background-color: ",
            [Function],
            ";
  border-radius: 1px;
  bottom: 0;
  position: fixed;
  top: 0;
  padding: 27px 32px 32px 32px;
  z-index: 1002;
  
  ",
            [Function],
            ";

  ",
            [Function],
            ";

  ",
            [Function],
            "
",
          ],
        },
        "defaultProps": Object {
          "theme": Object {
            "colors": Object {
              "base": "#00A376",
              "border": "#668491",
              "brand": "#00DC00",
              "destructive": Object {
                "hover": "#9F2C3F",
              },
              "error": "#C7384F",
              "focus": "#FFB500",
              "focusedIcon": "#335B6D",
              "focusedLinkBackground": "#FFDA7F",
              "info": "#0073C2",
              "primary": "#26A826",
              "secondary": "#006300",
              "success": "#00B000",
              "tertiary": "#004400",
              "warning": "#E96400",
              "white": "#FFFFFF",
              "whiteMix": "#E5F4E5",
            },
            "disabled": Object {
              "background": "#E5EAEC",
              "border": "#CCD6DA",
              "button": "#E5EAEC",
              "buttonText": "rgba(0,0,0,.2)",
              "disabled": "rgba(0,0,0,0.55)",
              "input": "#F2F4F5",
              "text": "rgba(0,0,0,0.3)",
            },
            "icon": Object {
              "focus": "#335B6D",
            },
            "name": "base",
            "shadows": Object {
              "cards": "0 3px 3px 0 rgba(0,20,29,0.2), 0 2px 4px 0 rgba(0,20,29,0.15)",
              "depth1": "0 5px 5px 0 rgba(0,20,29,0.2), 0 10px 10px 0 rgba(0,20,29,0.1)",
              "depth2": "0 10px 20px 0 rgba(0,20,29,0.2), 0 20px 40px 0 rgba(0,20,29,0.1)",
              "depth3": "0 10px 30px 0 rgba(0,20,29,0.1), 0 30px 60px 0 rgba(0,20,29,0.1)",
              "depth4": "0 10px 40px 0 rgba(0,20,29,0.04), 0 50px 80px 0 rgba(0,20,29,0.1)",
            },
            "text": Object {
              "color": "rgba(0,0,0,0.9)",
              "placeholder": "rgba(0,0,0,0.3)",
              "size": "14px",
            },
          },
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    open={true}
    position="left"
    size="extra-small"
    theme={
      Object {
        "colors": Object {
          "base": "#00A376",
          "baseBlue": "#255BC7",
          "border": "#668491",
          "brand": "#00DC00",
          "destructive": Object {
            "hover": "#9F2C3F",
          },
          "error": "#C7384F",
          "focus": "#FFB500",
          "focusedIcon": "#335B6D",
          "focusedLinkBackground": "#FFDA7F",
          "greyDarkBlue50": "#8099a4",
          "info": "#0073C2",
          "primary": "#00805D",
          "secondary": "#006045",
          "success": "#00B000",
          "tertiary": "#003F2E",
          "warning": "#FFB500",
          "white": "#FFFFFF",
          "whiteMix": "#E5F4E5",
        },
        "disabled": Object {
          "background": "#E5EAEC",
          "border": "#4d7080",
          "button": "#E5EAEC",
          "buttonText": "rgba(0,0,0,.2)",
          "disabled": "#b3c2c8",
          "input": "#1e499f",
          "text": "#003349",
        },
        "icon": Object {
          "focus": "#335B6D",
        },
        "name": "classic",
        "shadows": Object {
          "cards": "0 3px 3px 0 rgba(0,20,29,0.2), 0 2px 4px 0 rgba(0,20,29,0.15)",
          "depth1": "0 5px 5px 0 rgba(0,20,29,0.2), 0 10px 10px 0 rgba(0,20,29,0.1)",
          "depth2": "0 10px 20px 0 rgba(0,20,29,0.2), 0 20px 40px 0 rgba(0,20,29,0.1)",
          "depth3": "0 10px 30px 0 rgba(0,20,29,0.1), 0 30px 60px 0 rgba(0,20,29,0.1)",
          "depth4": "0 10px 40px 0 rgba(0,20,29,0.04), 0 50px 80px 0 rgba(0,20,29,0.1)",
        },
        "text": Object {
          "color": "rgba(0,0,0,0.9)",
          "placeholder": "rgba(0,0,0,0.3)",
          "size": "14px",
        },
      }
    }
  >
    <div
      className="c0"
      open={true}
      size="extra-small"
    />
  </StyledComponent>
</styled.div>
`;

exports[`SidebarStyle When classic style is passed to the component should render correct style 2`] = `
.c0 {
  cursor: pointer;
  position: absolute;
  right: 20px;
  top: 26px;
  z-index: 1;
  color: #668491;
  color: rgba(0,0,0,0.85);
  top: 15px;
}

.c0:hover {
  color: #335B6D;
}

.c0:hover {
  color: #255BC7;
}

<styled.div
  theme={
    Object {
      "colors": Object {
        "base": "#00A376",
        "baseBlue": "#255BC7",
        "border": "#668491",
        "brand": "#00DC00",
        "destructive": Object {
          "hover": "#9F2C3F",
        },
        "error": "#C7384F",
        "focus": "#FFB500",
        "focusedIcon": "#335B6D",
        "focusedLinkBackground": "#FFDA7F",
        "greyDarkBlue50": "#8099a4",
        "info": "#0073C2",
        "primary": "#00805D",
        "secondary": "#006045",
        "success": "#00B000",
        "tertiary": "#003F2E",
        "warning": "#FFB500",
        "white": "#FFFFFF",
        "whiteMix": "#E5F4E5",
      },
      "disabled": Object {
        "background": "#E5EAEC",
        "border": "#4d7080",
        "button": "#E5EAEC",
        "buttonText": "rgba(0,0,0,.2)",
        "disabled": "#b3c2c8",
        "input": "#1e499f",
        "text": "#003349",
      },
      "icon": Object {
        "focus": "#335B6D",
      },
      "name": "classic",
      "shadows": Object {
        "cards": "0 3px 3px 0 rgba(0,20,29,0.2), 0 2px 4px 0 rgba(0,20,29,0.15)",
        "depth1": "0 5px 5px 0 rgba(0,20,29,0.2), 0 10px 10px 0 rgba(0,20,29,0.1)",
        "depth2": "0 10px 20px 0 rgba(0,20,29,0.2), 0 20px 40px 0 rgba(0,20,29,0.1)",
        "depth3": "0 10px 30px 0 rgba(0,20,29,0.1), 0 30px 60px 0 rgba(0,20,29,0.1)",
        "depth4": "0 10px 40px 0 rgba(0,20,29,0.04), 0 50px 80px 0 rgba(0,20,29,0.1)",
      },
      "text": Object {
        "color": "rgba(0,0,0,0.9)",
        "placeholder": "rgba(0,0,0,0.3)",
        "size": "14px",
      },
    }
  }
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  cursor: pointer;
  position: absolute;
  right: 20px;
  top: 26px;
  z-index: 1;
  color: ",
            [Function],
            ";

  &:hover {
      color: ",
            [Function],
            ";
  };

  ",
            [Function],
            "
",
          ],
        },
        "defaultProps": Object {
          "theme": Object {
            "colors": Object {
              "base": "#00A376",
              "border": "#668491",
              "brand": "#00DC00",
              "destructive": Object {
                "hover": "#9F2C3F",
              },
              "error": "#C7384F",
              "focus": "#FFB500",
              "focusedIcon": "#335B6D",
              "focusedLinkBackground": "#FFDA7F",
              "info": "#0073C2",
              "primary": "#26A826",
              "secondary": "#006300",
              "success": "#00B000",
              "tertiary": "#004400",
              "warning": "#E96400",
              "white": "#FFFFFF",
              "whiteMix": "#E5F4E5",
            },
            "disabled": Object {
              "background": "#E5EAEC",
              "border": "#CCD6DA",
              "button": "#E5EAEC",
              "buttonText": "rgba(0,0,0,.2)",
              "disabled": "rgba(0,0,0,0.55)",
              "input": "#F2F4F5",
              "text": "rgba(0,0,0,0.3)",
            },
            "icon": Object {
              "focus": "#335B6D",
            },
            "name": "base",
            "shadows": Object {
              "cards": "0 3px 3px 0 rgba(0,20,29,0.2), 0 2px 4px 0 rgba(0,20,29,0.15)",
              "depth1": "0 5px 5px 0 rgba(0,20,29,0.2), 0 10px 10px 0 rgba(0,20,29,0.1)",
              "depth2": "0 10px 20px 0 rgba(0,20,29,0.2), 0 20px 40px 0 rgba(0,20,29,0.1)",
              "depth3": "0 10px 30px 0 rgba(0,20,29,0.1), 0 30px 60px 0 rgba(0,20,29,0.1)",
              "depth4": "0 10px 40px 0 rgba(0,20,29,0.04), 0 50px 80px 0 rgba(0,20,29,0.1)",
            },
            "text": Object {
              "color": "rgba(0,0,0,0.9)",
              "placeholder": "rgba(0,0,0,0.3)",
              "size": "14px",
            },
          },
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    theme={
      Object {
        "colors": Object {
          "base": "#00A376",
          "baseBlue": "#255BC7",
          "border": "#668491",
          "brand": "#00DC00",
          "destructive": Object {
            "hover": "#9F2C3F",
          },
          "error": "#C7384F",
          "focus": "#FFB500",
          "focusedIcon": "#335B6D",
          "focusedLinkBackground": "#FFDA7F",
          "greyDarkBlue50": "#8099a4",
          "info": "#0073C2",
          "primary": "#00805D",
          "secondary": "#006045",
          "success": "#00B000",
          "tertiary": "#003F2E",
          "warning": "#FFB500",
          "white": "#FFFFFF",
          "whiteMix": "#E5F4E5",
        },
        "disabled": Object {
          "background": "#E5EAEC",
          "border": "#4d7080",
          "button": "#E5EAEC",
          "buttonText": "rgba(0,0,0,.2)",
          "disabled": "#b3c2c8",
          "input": "#1e499f",
          "text": "#003349",
        },
        "icon": Object {
          "focus": "#335B6D",
        },
        "name": "classic",
        "shadows": Object {
          "cards": "0 3px 3px 0 rgba(0,20,29,0.2), 0 2px 4px 0 rgba(0,20,29,0.15)",
          "depth1": "0 5px 5px 0 rgba(0,20,29,0.2), 0 10px 10px 0 rgba(0,20,29,0.1)",
          "depth2": "0 10px 20px 0 rgba(0,20,29,0.2), 0 20px 40px 0 rgba(0,20,29,0.1)",
          "depth3": "0 10px 30px 0 rgba(0,20,29,0.1), 0 30px 60px 0 rgba(0,20,29,0.1)",
          "depth4": "0 10px 40px 0 rgba(0,20,29,0.04), 0 50px 80px 0 rgba(0,20,29,0.1)",
        },
        "text": Object {
          "color": "rgba(0,0,0,0.9)",
          "placeholder": "rgba(0,0,0,0.3)",
          "size": "14px",
        },
      }
    }
  >
    <div
      className="c0"
    />
  </StyledComponent>
</styled.div>
`;
